src/app/tests/TestSimpleSubscriptionResumptionStorage.cpp

```diff
@@ -0,0 +1,482 @@
+/*
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <lib/support/UnitTestRegistration.h>
+#include <nlunit-test.h>
+
+#if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
```

Comment:
Test should be enabled in gn based on flags.

maybe session resumption should be a separate lib that we may or may not depend on, then tests can always be run, it may just not be compiled in.

If this is a pain to setup though, we can just use the flag and please add a TODO.

Reply:
Filed #24564 to track moving this conditional compilation to GN system.